tests/filter.json

```diff
@@ -407,6 +407,23 @@
         {"a":"c", "d": "f"}
       ]
     },
+    {
+      "name": "exists and exists, data false",
+      "selector" : "$[?@.a&&@.b]",
+      "document" : [{"a": false, "b": false}],
+      "result": [
+        {"a": false, "b": false}
+      ]
+    },
+    {
+      "name": "exists or exists, data false",
+      "selector" : "$[?@.a||@.b]",
+      "document" : [{"a": false}, {"b": false}],
```

Comment:
Same here?

Reply:
Non-matching data has been added to this test case, plus data that matches one side of the _or_ expression.